core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -135,16 +135,7 @@
     private DataSize filterAndProjectMinOutputPageSize = DataSize.of(500, KILOBYTE);
```

Comment:
Even if number of rows after pushdown is smaller then without pushdown it could significantly increase cpu overhead  of underlying database (table scans might be much cheaper than join). I think it would be great to determine what's the impact of pushdown on underlying connectors. It could be that join pushdown is beneficial only when joins are very non selective.

Reply:
Agreed. Yet I would assume that you will still be able to disable pushdown on per-connector level in configuration. As well as per-query using session.

Reply:
Totally -- https://github.com/trinodb/trino/pull/6874 provides both catalog level config and session toggle.